modules/plottinggl/src/plotters/scatterplotgl.cpp

```diff
@@ -483,15 +592,94 @@ void ScatterPlotGL::objectPicked(PickingEvent* p) {
         } else {
             selectedIndices_.insert(id);
         }
+        selectedIndicesGLDirty_ = true;
+
         // selection changed, inform processor
         selectionChangedCallback_.invoke(selectedIndices_);
     }
+    p->setUsed(true);
 }
 
 uint32_t ScatterPlotGL::getGlobalPickId(uint32_t localIndex) const {
     return static_cast<uint32_t>(picking_.getPickingId(localIndex));
 }
 
+void ScatterPlotGL::dragRectChanged(const dvec2& start, const dvec2& end) {
+    if (properties_.dragRectSettings_.getMode() == DragRectangleSettingsInterface::Mode::None) {
+        return;
+    }
+    auto xbuf = xAxis_->getRepresentation<BufferRAM>();
+    // Selection should end up with elements inside of rect while 
+    // filtering should end up with elements outside of rect.
+    auto isFiltering =
+        properties_.dragRectSettings_.getMode() == DragRectangleSettingsInterface::Mode::Filtering;
+    auto filteringCompareOp = [](auto elem, auto tmin, auto tmax) -> bool {
+        return elem >= tmin && elem <= tmax;
+    };
+    auto selectionCompareOp = [](auto elem, auto tmin, auto tmax) -> bool {
+        return elem < tmin || elem > tmax;
+    };
+
+    std::unordered_set<size_t> selectedIndices =
+        xbuf->dispatch<std::unordered_set<size_t>, dispatching::filter::Scalars>(
+            [min = start[0], max = end[0], isFiltering = isFiltering,
+             selectionCompareOp = selectionCompareOp,
+             filteringCompareOp = filteringCompareOp](auto brprecision) {
```

Comment:
Use one lambda expression (based on `isFiltering`) instead to avoid if statement inside loop?

Reply:
Refactored this